proxy/m2_actions.c

```diff
@@ -1357,11 +1379,13 @@ _m2_container_snapshot(struct req_args_s *args, struct json_object *jargs)
 
 	meta1_urlv_shift_addr(urlv);
 	CLIENT_CTX(ctx, args, type, 1);
+	char *url = _resolve_service_id(urlv[0]);
 	GByteArray * _pack(const struct sqlx_name_s *n) {
-		return sqlx_pack_SNAPSHOT(n, urlv[0], target_cid, seq_num, DL());
+		return sqlx_pack_SNAPSHOT(n, url, target_cid, seq_num, DL());
 	}
 
 	err = _resolve_meta2(args, CLIENT_PREFER_MASTER, _pack, NULL);
+	free(url);
```

Comment:
Mixing GLib-2.0 and glibc memory allocation functions is generally not a good idea.